docs/api/field_type_type_and_value.md

```diff
@@ -0,0 +1,166 @@
+
+In contrast to `acceptValue()` this method validates the plausibility of the given value, based on the Field Type settings and validator configuration, stored in the corresponding `FieldDefinition`.
+
+### Serialization
+
+When [REST API](rest_api_guide.md) is used, conversion needs to be done for Field Type values, settings and validator configurations. These are converted to and from a simple hash format that can be encoded in REST payload. As conversion needs to be done both when transmitting and receiving data through REST, Field Type implements following pairs of methods:
+
+|Method|Description|
+|------|-----------|
+|`toHash()`|Converts Field Type Value into a plain hash format.|
+|`fromHash()`|Converts the other way around.|
+|`fieldSettingsToHash()`|Converts Field Type settings to a simple hash format.|
+|`fieldSettingsFromHash()`|Converts the other way around.|
+|`validatorConfigurationToHash()`|Converts Field Type validator configuration to a simple hash format.|
+|`validatorConfigurationFromHash()`|Converts the other way around.|
+
+## Registration
+
+A Field Type must be registered as a service.
+
+``` yaml
+services:
+    EzSystems\EzPlatformMatrixFieldtype\FieldType\Type:
+        parent: ezpublish.fieldType
+        tags:
+            - {name: ezpublish.fieldType, alias: 'ezmatrix'}
+```
+
+`parent`
+
+As described in the [Symfony Dependency Injection Component documentation](http://symfony.com/doc/master/components/dependency_injection/parentservices.html), the `parent` config key indicates that you want your service to inherit from the parent's dependencies, including constructor arguments and method calls. This helps avoiding repetition in your Field Type configuration and keeps consistency between all Field Types.
+
+`tags`
+
+You must tag the Field Type service with `ezpublish.fieldType` so it is recognized as a regular Field Type.
+The `alias` key is the `fieldTypeIdentifier`.
+
+!!! tip
+
+    Configuration of built-in Field Types is located in [`EzPublishCoreBundle/Resources/config/fieldtypes.yml`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.0/eZ/Publish/Core/settings/fieldtypes.yml).
+
+## Field Type settings
+
+It is recommended to use a simple associative array format for the settings schema returned by `eZ\Publish\SPI\FieldType\FieldType::getSettingsSchema()`, which follows these rules:
+
+- The key of the associative array identifies a setting (e.g. `default`)
+- Its value is an associative array describing the setting using:
+    - `type` to identify the setting type (e.g. `int` or `string`)
+    - `default` containing the default setting value
+
+An example schema could look like this:
+
+``` php
+[
+    'backupData' => [
+        'type' => 'bool',
+        'default' => false
+    ],
+    'defaultValue' => [
+        'type' => 'string',
+        'default' => 'Default Value'
+    ]
+];
+```
+
+The settings are mapped into Symfony forms via the [FormMapper](field_type_form_and_template.md#formmapper).
```

Comment:
Ping @alongosz to clarify where and why do we need `getSettingsSchema`, because as far as I know these are not mapped to anything and not needed as long as FieldDefinitionFormMapper::mapFieldDefinitionForm provides correct Symfony Forms fields. Maybe deprecation candidate?

Reply:
> Ping @alongosz to clarify where and why do we need `getSettingsSchema`, because as far as I know these are not mapped to anything and not needed as long as `FieldDefinitionFormMapper::mapFieldDefinitionForm` provides correct Symfony Forms fields. Maybe deprecation candidate?

No idea. It requires deeper investigation.